### PR TITLE
Upgrade Betterment/plaid-java to support TLSv1.1 and TLSv1.2

### PR DESCRIPTION
**Pull request by superwomyn**
https://trello.com/c/U9YaXI23/2752-2-upgrade-plaid-java-and-ruby-clients-to-support-tls-1-1

/domain @ceslami @adamlangsner @reustmd @samandmoore 
/platform @ceslami @adamlangsner @reustmd @samandmoore

### CODE REVIEW

File: src/main/java/com/plaid/client/http/ApacheHttpClientHttpDelegate.java [hunk 1]
```diff
@@ -52,20 +57,49 @@ public ApacheHttpClientHttpDelegate(String baseUri, CloseableHttpClient httpClie
...
+        SSLConnectionSocketFactory sf = new SSLConnectionSocketFactory(
+                sslContext,
+                new String[] { "TLSv1.1", "TLSv1.2" },
+                null,
+                SSLConnectionSocketFactory.getDefaultHostnameVerifier());
+
+        return HttpClients
+                .custom()
+                .setSSLSocketFactory(sf);
+    }
```
Comment (samandmoore) [inline src/main/java/com/plaid/client/http/ApacheHttpClientHttpDelegate.java:93]: FYI to everyone else reviewing this code, @superwomyn and I paired on this chunk here. In order to land on this somewhat interesting-looking configuration, we just scraped the innards out of this method: `HttpClients.createDefault()` and created out own `SSLConnectionSocketFactory` (also by scraping the innards out of the underlying defaults method.

**Comment by samandmoore** [comment]
<< domain lgtm. i'm going to give @ceslami and @adamlangsner a bit if they wanna grab the review reigns, otherwise i will come back and give this another lg.

**Comment by ceslami** [comment]
I can't give platform on this, but looks good!

**Comment by samandmoore** [comment]
<< platform lgtm

**Pull request merged**